### PR TITLE
fix: evaluate prerequisite features correctly

### PR DESCRIPTION
**Pull request by ono-max**

### CODE REVIEW

**Comment by cre8ivejp** [comment]
@ono-max, can you also implement it for the rest server?
Because we will release the prerequisite feature before we release that go-server sdk with the grpc implementation.

File: pkg/gateway/api/api_grpc.go [hunk 1]
```diff
@@ -393,11 +388,47 @@ func (s *grpcGatewayService) GetEvaluation(
...
 	}, nil
 }
 
+func (s *grpcGatewayService) getTargetFeatures(fs []*featureproto.Feature, id string) ([]*featureproto.Feature, error) {
+	feature, err := s.findFeature(fs, id)
+	if err != nil {
+		return nil, err
+	}
+	if len(feature.Prerequisites) > 0 {
+		return fs, nil
```
Comment (masaaania) [inline pkg/gateway/api/api_grpc.go:406]: question:

Is it possible to select only the prerequisite feature flags?

Comment (ono-max) [reply pkg/gateway/api/api_grpc.go:408]: Talked at meeting.

File: pkg/gateway/api/api.go [hunk 2]
```diff
@@ -348,14 +343,53 @@ func (s *gatewayService) getEvaluation(w http.ResponseWriter, req *http.Request)
...
 }
 
+func (s *gatewayService) getTargetFeatures(fs []*featureproto.Feature, id string) ([]*featureproto.Feature, error) {
+	feature, err := s.findFeature(fs, id)
+	if err != nil {
+		return nil, err
+	}
+	if len(feature.Prerequisites) > 0 {
+		// If we select only the prerequisite feature flags, we have to get them recursively.
+		// Thus, we evaluate all features here to avoid complex logic.
```
Comment (masaaania) [inline pkg/gateway/api/api.go:366]: 👍

**Review by masaaania — APPROVED** [review]
lgtm!

**Review by cre8ivejp — APPROVED** [review]
Nice!

**Pull request merged**